vms/fx/factory.go

```diff
@@ -0,0 +1,9 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fx
+
+// Factory returns an instance of a feature extension
+type Factory interface {
+	New() any
```

Comment:
Another idea is to return an interface here, but we're already currently returning the `any` type so I'm going to punt this off to a follow-up since my focus is to currently just refactor the vm code.